src/simplify_dynamicquantizelinear.cpp

```diff
@@ -0,0 +1,148 @@
+                                    match::name("nearbyint")(
+                                        match::arg(0)(match::name("clip").bind("saturate")))
+                                        .bind("round")))
+                                .bind("convert"))))
+                        .bind("quant_lin"))))
+                .bind("target"));
+    }
+
+    void apply(module& m, const match::matcher_result& r) const
+    {
+        auto target_op = r.instructions["target"];
+        /* Need to modify the uint8 min/max range as well as final convert to convert to int8 */
+        auto convert_op = r.instructions["convert"];
+        // Ops to get q_min/q_max quickly
+        auto round_op    = r.instructions["round"];
+        auto quant_op    = r.instructions["quant_lin"];
+        auto saturate_op = r.instructions["saturate"];
+        auto q_min       = saturate_op->inputs().at(1);
+        auto q_max       = saturate_op->inputs().at(2);
+
+        // get new desired range defined by int8_t
+        const auto x_min = std::numeric_limits<int8_t>::min();
+        const auto x_max = std::numeric_limits<int8_t>::max();
+
+        // Replace min/max of uint8 with min/max of int8 - q_range is identical so doesn't need to
+        // be modified. Need to replace other ops which also take uint8 values first.
+        auto x_type     = q_min->get_shape().type();
+        auto q_min_int8 = m.add_literal(
+            migraphx::literal{migraphx::shape{x_type, q_min->get_shape().lens()}, {x_min}});
+        auto q_max_int8 = m.add_literal(
+            migraphx::literal{migraphx::shape{x_type, q_max->get_shape().lens()}, {x_max}});
+
+        m.replace_instruction(q_min, q_min_int8);
+        m.replace_instruction(q_max, q_max_int8);
+
+        auto new_conv = m.insert_instruction(
+            convert_op,
+            migraphx::make_op("convert", {{"target_type", migraphx::shape::int8_type}}),
+            round_op);
+
+        // Convert inputs to the target op to ensure we're all int8 as part of our splice
+        // This will be optimized out as part of simplify_reshapes if convert is redundant
+        auto inputs = target_op->inputs();
+        std::vector<instruction_ref> converted_inputs;
+        for(auto& in : inputs)
+        {
+            auto item = in;
+            if(in->get_shape().type() == migraphx::shape::uint8_type)
+            {
+                item = m.insert_instruction(
+                    target_op,
+                    migraphx::make_op("convert", {{"target_type", migraphx::shape::int8_type}}),
+                    in);
+            }
+            converted_inputs.push_back(item);
+        }
+
+        auto new_target_op =
+            m.insert_instruction(target_op, target_op->get_operator(), converted_inputs);
+
+        auto data  = quant_op->inputs().at(0);
+        auto scale = quant_op->inputs().at(1);
+
+        m.replace_instruction(target_op, new_target_op);
+        m.remove_instruction(target_op);
```

Comment:
Dont call `remove_instruction`. DCE will remove unused instructions. There can be problems with removing while iterating(such as infintie loops, etc). The DCE pass will do this correctly so we should just rely on it to remove the unused instructions.

Reply:
What alternative would you suggest here then? A replace will still call compute_shapes() and then propagate through and generate an issue with dot/conv on the same_type() check

Reply:
You need to be replacing the dot/conv instead.